src/library-authoring/components/ComponentCard.tsx

```diff
@@ -35,16 +34,15 @@ const ComponentCard = ({ hit }: ComponentCardProps) => {
     showOnlyPublished ? formatted.published?.displayName : formatted.displayName
   ) ?? '';
 
-  const { navigateTo } = useLibraryRoutes();
   const selectComponent = useCallback(() => {
     if (!componentPickerMode) {
-      navigateTo({ selectedItemId: usageKey });
+      openItemSidebar(usageKey, SidebarBodyItemId.ComponentInfo);
     } else {
       // In component picker mode, we want to open the sidebar
       // without changing the URL
       openComponentInfoSidebar(usageKey);
     }
-  }, [usageKey, navigateTo, openComponentInfoSidebar]);
+  }, [usageKey, openItemSidebar, openComponentInfoSidebar, componentPickerMode]);
```

Comment:
@DanielVZ96 `componentPickerMode` is necessary here?

Reply:
It was added automatically by my linter because it's used above inside the callback. but it's most probably not necessary because I don't think it should change after the initial load. if you think we should remove it i can remove it.